### PR TITLE
feat: update mlflow version to support newer version of protobuf

### PR DESCRIPTION
**Pull request by khorshuheng**
**What this PR does / why we need it**:

* Mlflow 1.26.1 or above is required in order to be compatible with newer version of protobuf.

* This PR also drops the support for Python 3.7 from the real-time and batch predictor servers (continuing from https://github.com/caraml-dev/merlin/pull/464). Docs and examples are updated accordingly.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Upgrade MlFlow client version; drop Python 3.7 support 
```

**Checklist**

- [ ] Added unit test, integration, and/or e2e tests
- [ ] Tested locally
- [x] Updated documentation
- [ ] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

**Comment by khorshuheng** [comment]
@krithika369 @ariefrahmansyah Release note for Mlflow 1.26.1 : https://github.com/mlflow/mlflow/releases/tag/v1.26.1

This should allow us to use newer version of protobuf.

File: python/batch-predictor/requirements.txt [hunk 1]
```diff
@@ -1,8 +1,6 @@
 findspark
 pyspark==3.0.1
-mlflow>=1.2.0,<=1.23.0
+mlflow>=1.26.1,<=2.0.0
 cloudpickle==2.0.0
 pyarrow>=0.14.1,<=9.0.0
-protobuf>=3.0,<4.0.0
```
Comment (krithika369) [inline python/batch-predictor/requirements.txt:6]: Should we continue to pin protobuf version? Or is it expected that the new mlflow sdk is compatible with protobuf 4.X ?

Comment (khorshuheng) [reply python/batch-predictor/requirements.txt:6]: According to the release note for Mlflow 1.26.1, it is expected to be compatible with protobuf 4.X. Although, i suppose we can pin it to <5.0.0 .

File: python/pyfunc-server/docker/Dockerfile [hunk 2]
```diff
@@ -24,7 +24,7 @@ WORKDIR /pyfunc-server
 RUN if [ ! -z "$GOOGLE_APPLICATION_CREDENTIALS" ]; then gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS}; fi
 RUN gsutil cp -r ${MODEL_URL} .
 RUN /bin/bash -c ". activate merlin-model && \
-    sed -i 's/mlflow$/mlflow==1.23.0/' model/conda.yaml && \
+    sed -i 's/mlflow$/mlflow==1.26.1/' model/conda.yaml && \
```
Comment (ariefrahmansyah) [inline python/pyfunc-server/docker/Dockerfile:27]: Any particular reason to not use the same sed args as we have for batch predictor above (`sed -i 's/\bmlflow[^[:space:]]*/mlflow==1.26.1/'`)?

File: python/sdk/Dockerfile [hunk 3]
```diff
@@ -12,5 +12,5 @@ COPY . .${WORKDIR}
 
 ARG VERSION
```
Comment (ariefrahmansyah) [inline python/sdk/Dockerfile:13]: nit: this arg is not used anymore. we can remove it here and also on build command in github action jobs.

**Review by ariefrahmansyah — APPROVED** [review]
Overall, LGTM with nit comments. Thanks for updating MLflow on Merlin @khorshuheng!

File: python/pyfunc-server/docker/Dockerfile (hunk 2 above)
Comment (khorshuheng) [reply python/pyfunc-server/docker/Dockerfile:27]: Yes, it should have been the same.

File: python/batch-predictor/requirements.txt [hunk 4]
```diff
@@ -1,8 +1,6 @@
 findspark
 pyspark==3.0.1
-mlflow>=1.2.0,<=1.23.0
+mlflow>=1.26.1,<=2.0.0
```
[the remark this answers is not in the document]
Comment (khorshuheng) [reply python/batch-predictor/requirements.txt:3]: Typo on my end, should have been less than instead of less than or equals.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -47,8 +47,6 @@ jobs:
       fail-fast: false
       matrix:
         python-version: ["3.8", "3.9", "3.10"]
-    needs:
```
Comment (krithika369) [inline .github/workflows/release.yml:50]: @khorshuheng, is this change intentional or something that was done temporarily, to allow the testing flow to proceed?

Comment (khorshuheng) [reply .github/workflows/release.yml:50]: This was done to allow testing flow to proceed.
